### ISSUE TITLE
Use 'gemspec :development_group => :test' by default

### ISSUE THREAD

**Issue by e2** [bundler/bundler#3303]
##### TL;DR - I'd strongly suggest `gemspec development_group: :test` as the default (it looks a bit confusing, but it really does what people generally want).

First, all the confusing kinds of deps we have:

1) "all deps" - everything (which is what Bundler resolves)
2) "gem deps" - deps for users to install and use gems
3) "gem dev deps" - for `gem install --dev` - though typically build tools and test frameworks
4) "bundler development group" - typically installation skipped through bundler's `--without` option
5) "bundler test group" - normally would match the "gem dev deps", but without gem build tools (unless releasing gems from Travis/CI)
6) "bundler default group" - basically, everything not in "test" or "development"
7) "rails development group" - unrelated to the issue (!), because we're talking about gems, not apps

And here are the kinds of deps we really want:

1) "gem deps" (always needed)
2) "test deps" (without build tools, unless building gem from Travis/CI)
3) "build deps" (extra minimum set of tools for locally building a valid gem)
4) "human deps" (stuff only human beings may find useful, e.g. Pry+plugins, Guard+plugins, Livereload, servers, manual testing, UI design)

And here's how those four groups would "ideally" be set up:

1) "gem deps" -> gemspec: "gem deps", Gemfile: `gemspec` keyword
2) "test deps" -> gemspec: "gem dev deps", Gemfile: only through `development_group: :test`
3) "build deps" -> gemspec: nothing here (unless building gem on CI), Gemfile: any group except `development` and `test`
4) "human deps" -> gemspec: nothing here, Gemfile: `group :development`

Key realizations:
- it doesn't make sense for `gem install --dev` to install "everything" since Bundler has become widespread
- Bundler allows separating "test deps" from everything else (which dramatically speeds up CI setup)
- "test deps" have become more important than "dev deps", while CIs have become "testers" who don't need the dev tools
- this doesn't affect app users at all (and in their world "development" and "test" pretty much overlap anyway)
- in apps, Gemfiles are for app deps - but in gems, they are purely for development (and building if not done on CI)

What this breaks:
- absolutely nothing because in the worst case scenario (bad setup) "gem dev deps" will be installed on a test environment - which may break builds, but for a very good reason (and the line in the Gemfile will show that it's on purpose).
